docs/ConcurrencyRoadmap.md

```diff
@@ -0,0 +1,265 @@
+
+Features expected to be discussed as part of a second phase include:
+
+- introduction of an `actorlocal` restriction on types;
+- compiler support for guaranteed correct "copy on write" types via a `mutableIfUnique` class type;
+- attributes to opt-out of actor isolation, for example where thread safety is being handled through some other means.
+
+## Glossary of Concepts
+
+These are the basic concepts that will be used throughout the design, briefly defined here.
+
+* A **synchronous function** is the kind of function that Swift programmers are already used to: it runs to completion on a single thread, with no interleaving code except for any synchronous functions it calls.
+* A **thread** refers to the underlying platform’s concept of threads. Platforms vary, but tend to share basic characteristics: true concurrency requires creating a platform thread, but creating and running platform threads is expensive. C function calls, and ordinary synchronous Swift functions, require the use of a platform thread.
+* An **asynchronous function** is a new kind of function which does not have to run to completion without interruption. An interruption causes the function to be **suspended**. The point at which an async function may give up its thread is a **suspension point**.
+* A **task** is an operation that runs asynchronously. All asynchronous functions run as part of some task.  When an asynchronous function calls another asynchronous function, the call is still part of the the same task, even if the call has to change actors. A task is the analogue of a thread for asynchronous functions.
+* An async function can create a **child task**. Child tasks inherit some of the structure of their parent task, including its priority, but can run concurrently with it.  However, this concurrency is bounded: a function that creates a child task must wait for it to end before returning.
+* A program wishes to initiate independent concurrent work that can outlast its spawning context uses a **detached task** rather than a bounded child task.
+* A **partial task** is a unit of schedulable work. When the currently-executing function in a task is suspended, that is the end of the partial task, and a new partial task is created to continue the overall task’s work.
+* An **executor** is a service which accepts the submission of partial tasks and arranges for some thread to run them. An async function that is currently running always knows the executor on which it's running. An executor is called **exclusive** if the partial tasks submitted to it will never be run concurrently.
+* An **actor** is an independent part of the program which can run code.  It can only run one piece of code at a time — that is, it acts as an exclusive executor — but the code it runs can execute concurrently with the code run by other actors.
```

Comment:
```suggestion
* An **actor** is an independent part of the program which can run code. It can only run one piece of code at a time — that is, it acts as an exclusive executor — but the code it runs can execute concurrently with the code run by other actors.
```